src/components/FlawForm.vue

```diff
@@ -97,6 +98,22 @@ const flawCvss3CaculatorLink = computed(
 const onReset = () => {
   flaw.value = deepCopyFromRaw(initialFlaw.value as Record<string, any>) as ZodFlawType;
 };
+
+function validateFlawEmbargoDates(flaw) {
+  if (flaw.embargoed) {
+    // if embargoed and updated date is older than now, it shows an error to set the current date or a future date instead
+    if (DateTime.fromISO(flaw.unembargo_dt as string).toISODate() < DateTime.now().toISODate()) {
+      return errors.embargo_old_public_dt;
+    }
+  } else {
+    // if NOT embargoed and updated date is in the future, it shows an error, to set the current date or an older date instead
+    if (DateTime.fromISO(flaw.unembargo_dt as string).toISODate() > DateTime.now().toISODate()) {
+      return errors.unembargo_future_public_dt;
+    }
```

Comment:
There is also one additional case which is apparently not mentioned in the Jira, but should be probably added. When Flaw is not embargoed and the unembargo_dt is not set, this is also invalid case and user should be prompted to set the unembargo_dt correctly. Also just a note, all these things are of course validated on the OSIDB side, so this servers merely as a convenience for user so he can observe these errors earlier.

Reply:
What happens when is embargoed and dt is not set?

I have just pushed this commit:

https://github.com/RedHatProductSecurity/osim/pull/105/commits/4c26ecd4d451f2fd695dfc844f4a78281918831f#diff-bb5d5f91b338ff8b35ca76429e074f4af4afcd509d5adf9d6004016b419fd6cdR108

https://github.com/RedHatProductSecurity/osim/pull/105/commits/4c26ecd4d451f2fd695dfc844f4a78281918831f#diff-967a205363234f686e2f6df964ed2631bc6e26c2ba4f9d69abad87898d0d3461R443

Does it solves the use case?

Reply:
No, I don't think so.
Flaw is embargoed and unembargo_dt is not set -> valid state
Flaw is not embargoed and unembargo_dt is not set -> invalid state